jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -405,7 +405,16 @@ public Collection<Resource> getAllResources()
             boolean noDepth = true;
 
             for (Iterator<Resource> i = children.iterator(); noDepth && i.hasNext(); )
-                noDepth = !i.next().isDirectory();
+            {
+                Resource resource = i.next();
+                if (resource.isDirectory())
+                {
+                    // If the directory is a symlink we do not want to go any deeper.
```

Comment:
why?

Reply:
This prevents symlink loops, and is the pre-existing behavior.